src/components/provisioning/provisioning.js

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { ProgressBar } from 'patternfly-react';
+import get from 'lodash.get';
+import { noop } from 'patternfly-react';
+import { connect } from '../../redux';
+import { manageMiddlewareServices, mockMiddlewareServices } from '../../services/middlewareServices';
+import { DEFAULT_SERVICES } from '../../common/serviceInstanceHelpers';
+
+const PROVISION_SERVICES = [
+  DEFAULT_SERVICES.AMQ,
+  DEFAULT_SERVICES.ENMASSE,
+  DEFAULT_SERVICES.CHE,
+  DEFAULT_SERVICES.LAUNCHER,
+  DEFAULT_SERVICES.FUSE
+]
+
+class Provisioning extends React.Component {
+  componentDidMount() {
+    const { manageWalkthroughServices, mockWalkthroughServices } = this.props;
+    if (window.OPENSHIFT_CONFIG.mockData) {
+      mockWalkthroughServices(window.OPENSHIFT_CONFIG.mockData);
+      return;
+    }
+    manageWalkthroughServices();
+  }
+
+  static getMiddlwareServiceProgress(services) {
+    const svcNames = Provisioning.getProvisionedMiddlewareServices(services);
+    return (svcNames.length / PROVISION_SERVICES.length) * 100;
+  }
+
+  static getProvisionedMiddlewareServices(services) {
+    return PROVISION_SERVICES.map(svcName => {
+      const svc = Provisioning.getServiceInstanceByClassName(services, svcName);
+      if (!svc || !Provisioning.isMiddlewareServiceProvisioned(svc)) {
+        return null;
+      }
+      return svc;
+    }).filter(svc => !!svc);
+  }
+
+  static areMiddlewareServicesReady(services) {
+    for (let svcName of PROVISION_SERVICES) {
+      const svc = Provisioning.getServiceInstanceByClassName(services, svcName);
+      if (!svc || !Provisioning.isMiddlewareServiceProvisioned(svc)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  static getServiceInstanceByClassName(services, classToFind) {
+    return services.find(svc => get(svc, 'spec.clusterServiceClassExternalName') === classToFind);
+  }
+
+  static isMiddlewareServiceProvisioned(svc) {
+    if (svc.status && svc.status.conditions && svc.status.conditions[0]) {
+      return svc.status.conditions[0].status === 'True';
+    }
+    return false;
+  }
+
+  // NOTE: All styling is inline as it'll all be removed anyway.
+  loadingScreen(services) {
+    const provisionProgress = Provisioning.getMiddlwareServiceProgress(services);
+    return (
+      <div style={{marginLeft: 'auto', marginRight: 'auto', display: 'block', textAlign: 'center', maxWidth: '800px', marginTop: '150px'}}>
```

Comment:
@aidenkeating can you create a class for this and attach these attributes to that? That will make it easier for future revisions, to just have to change a CSS file. Thanks!